src/wgpu_engine.rs

```diff
@@ -553,7 +552,7 @@ impl WgpuEngine {
                     let src_buf = self
                         .bind_map
                         .get_gpu_buf(proxy.id)
-                        .ok_or("buffer not in map")?;
+                        .ok_or(VelloError::UnavailableBufferUsed(proxy.name, "download"))?;
```

Comment:
This error doesn't feel meaningful... is there a way for the developer to react if this is triggered? I don't think so... this error would only happen if we (as Vello maintainers) aren't handling the buffers correctly. In such case, I'd accept a panic here instead with a safety comment.

Reply:
This is a question about where our abstraction boundaries lie. Concretely, there are effectively two parts:
1) The recording abstraction
2) Connecting that abstraction to the Vello shaders

As seen in #416, the recording can theoretically be used for additional functionality by end-users.
Our current API here is very much research quality, and so. I think it's likely that we'll want to have slightly more fine-grained error types for this in future (i.e. a different error at the recording/higher-level Vello levels), but this is fine for this PR.